var/spack/repos/builtin/packages/namd/package.py

```diff
@@ -304,6 +315,9 @@ def edit(self, spec, prefix):
             if "+single_node_gpu" in spec:
                 opts.extend(["--with-single-node-hip"])
 
+        if "+memopt" in spec:
```

Comment:
To be in line with what's being merged in #46126 and similar PRs:
```suggestion
        if spec.satisfies("+memopt"):
```

Reply:
Changes pushed as suggested.